src/Http/Headers/src/SetCookieHeaderValue.cs

```diff
@@ -554,12 +580,13 @@ public override bool Equals(object obj)
                 && StringSegment.Equals(Path, other.Path, StringComparison.OrdinalIgnoreCase)
                 && Secure == other.Secure
                 && SameSite == other.SameSite
-                && HttpOnly == other.HttpOnly;
+                && HttpOnly == other.HttpOnly
+                && HeaderUtilities.AreEqualCollections(Extensions, other.Extensions);
```

Comment:
Does this consider order of extensions?

Reply:
Changed that to consider order of extensions

Reply:
Sorry, I meant the opposite, order shouldn't matter.

Reply:
Oh, I was thinking what you meant by that and chose the wrong option. I have reverted it back and changed the test for it.

Reply:
Yeah, I just meant to ask about the behavior of HeaderUtilities.AreEqualCollections.

Reply:
Just to confirm things. The HeaderUtilities.AreEqualCollections is indeed checking if two unordered list have same content. It does it ing O(n*m) time, but since the number of extensions is not likely to reach any high number, so it doesn't really matter.